src/Editor.elm

```diff
@@ -413,12 +378,24 @@ viewNodeInputs additional z =
             :: viewRuleType z
             :: div [ class "onclick-menu change", tabindex 0 ]
                 [ ul [ class "onclick-menu-content" ]
-                    [ li [] [ button [ onClick (ChangeToAlpha z) ] [ text "Change to α" ] ]
-                    , li [] [ button [ onClick (ChangeToBeta z) ] [ text "Change to β" ] ]
-                    , li [] [ button [ onClick (ChangeToGamma z) ] [ text "Change to γ" ] ]
-                    , li [] [ button [ onClick (ChangeToDelta z) ] [ text "Change to δ" ] ]
-                    , li [] [ button [ onClick (ChangeToRefl z) ] [ text "Change to Reflexivity" ] ]
-                    , li [] [ button [ onClick (ChangeToLeibnitz z) ] [ text "Change to Leibnitz" ] ]
+                    [ li [] [ button [ onClick (ChangeToUnary Alpha z) ] [ text "Change to α" ] ]
+                    , li [] [ button [ onClick (ChangeToBinary Beta z) ] [ text "Change to β" ] ]
+                    , li [] [ button [ onClick (ChangeToUnaryWithSubst Gamma z) ] [ text "Change to γ" ] ]
+                    , li [] [ button [ onClick (ChangeToUnaryWithSubst Delta z) ] [ text "Change to δ" ] ]
+                    , li [] [ button [ onClick (ChangeToUnary Refl z) ] [ text "Change to Reflexivity" ] ]
+                    , li [] [ button [ onClick (ChangeToUnary Leibnitz z) ] [ text "Change to Leibnitz" ] ]
+                    , li [] [ button [ onClick (ChangeToUnary MP z) ] [ text "Change to MP" ] ]
+                    , li [] [ button [ onClick (ChangeToUnary MT z) ] [ text "Change to MT" ] ]
+                    , li [] [ button [ onClick (ChangeToBinary Cut z) ] [ text "Change to Cut" ] ]
+                    , li [] [ button [ onClick (ChangeToUnary HS z) ] [ text "Change to HS" ] ]
+                    , li [] [ button [ onClick (ChangeToUnary DS z) ] [ text "Change to DS" ] ]
+                    , li [] [ button [ onClick (ChangeToUnary NCS z) ] [ text "Change to NCS" ] ]
+                    , li [] [ button [ onClick (ChangeToBinary ECDF z) ] [ text "Change to ECDF" ] ]
+                    , li [] [ button [ onClick (ChangeToBinary ECDT z) ] [ text "Change to ECDT" ] ]
+                    , li [] [ button [ onClick (ChangeToUnary ESFF z) ] [ text "Change to ESFF" ] ]
+                    , li [] [ button [ onClick (ChangeToUnary ESFT z) ] [ text "Change to ESFT" ] ]
+                    , li [] [ button [ onClick (ChangeToUnary ESTF z) ] [ text "Change to ESTF" ] ]
+                    , li [] [ button [ onClick (ChangeToUnary ESTT z) ] [ text "Change to ESTT" ] ]
```

Comment:
Please, define some auxiliary functions that you can reuse here in this Change menu and the Add menu below to make this code cleaner.

I think there could be a single `ruleMenu` function that will be called here as:
```elm
ruleMenu ChangeToUnary ChangeToUnaryWithSubst ChangeToBinary "Change to" "change" z
```
and below as
```elm
ruleMenu ExpandUnary ExpandUnaryWithSubst ExpandBinary "Add" "add" z
```
and it will generate all menu items with appropriate messages and labels.

I imagine it roughly like this:
```elm
ruleMenu unaryMsg unaryWithSubstMsg binaryMsg labelPrefix cls z =
    let
        unaryItem ext = menuItem (unaryMsg ext z) (labelPrefix ++ " " ++ unaryExtTypeName ext)
        unaryWithSubstItem ext = …
        binaryItem ext = …
    in
        div [ class … ++ cls, … ]
            [ ul [ class … ] <|
                [ unaryItem Alpha
                , binaryItem Beta
                …
                ]
                …
                ++ List.map unaryItem [ ESFF, …, ESTT ]
        …
```
where `menuItem : Msg -> String -> Html` makes the `li` with `button`, and `unaryExtTypeName : UnaryExtType -> String` etc. yield a name of a rule.

Reply:
A side note: These are pure CSS menus, which is nice. However, the “button” that activates this menu is made as a `::before` pseudoelement and its label (“Change” or “Add”) is also generated in CSS as the content of this pseudoelement depending on the menu’s class. This is a very ugly solution. I would strongly prefer to have a real `button` (or at least `span`) element inside the menu `div` before the `ul`, which would contain a `text` with the button’s label.